File: src/main/java/ru/ioleksiv/telegram/bot/api/annotations/filter/secondary/OrderInfoFilter.java
package ru.ioleksiv.telegram.bot.api.annotations.filter.secondary;

import ru.ioleksiv.telegram.bot.api.annotations.filter.primitive.StringFilter;
import ru.ioleksiv.telegram.bot.api.model.annotation.AnnotationState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface OrderInfoFilter {

    AnnotationState value() default AnnotationState.ON;

    StringFilter name() default @StringFilter(AnnotationState.OFF);

    StringFilter phoneNumber() default @StringFilter(AnnotationState.OFF);

    StringFilter email() default @StringFilter(AnnotationState.OFF);

    ShippingAddressFilter shippingAddress() default @ShippingAddressFilter(AnnotationState.OFF);

}

File: core/src/main/kotlin/io/github/oleksivio/tl/kbot/core/annotations/filter/composite/OrderInfoFilter.kt
package io.github.oleksivio.tl.kbot.core.annotations.filter.composite

import io.github.oleksivio.tl.kbot.core.annotations.filter.primitive.StringFilter
import io.github.oleksivio.tl.kbot.core.model.annotation.AnnotationState

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class OrderInfoFilter(
    val status: AnnotationState = AnnotationState.ON,
    val validator: Array<String> = [],
    val name: StringFilter = StringFilter(
        status = AnnotationState.OFF
    ),
    val phoneNumber: StringFilter = StringFilter(
        status = AnnotationState.OFF
    ),
    val email: StringFilter = StringFilter(
        status = AnnotationState.OFF
    ),
    val shippingAddress: ShippingAddressFilter = ShippingAddressFilter(
        status = AnnotationState.OFF
    )
)
